pkg/services/pluginsintegration/clientmiddleware/tracing_header_middleware.go

```diff
@@ -102,3 +105,25 @@ func (m *TracingHeaderMiddleware) RunStream(ctx context.Context, req *backend.Ru
 	m.applyHeaders(ctx, req)
 	return m.BaseHandler.RunStream(ctx, req, sender)
 }
+
+// sanitizeGrpcHeaderValue sanitizes header values according to HTTP/2 gRPC specification.
+// The spec defines ASCII-Value as 1*( %x20-%x7E ) ; space and printable ASCII
+// Control characters (0x00-0x1F) are percent-encoded.
+// Allows printable ASCII (0x20-0x7E) and extended characters (> 0x7F).
+func sanitizeGrpcHeaderValue(value string) string {
```

Comment:
Nit: WDYT about renaming this `santizeHTTPHeaderValueForGRPC`? My first thought reading this was that we were dealing with gRPC back to HTTP, rather than the other way around!

Reply:
Fixed in https://github.com/grafana/grafana/pull/107339/commits/c3536040f3adaae80b21d1de73d87da1c96bce88